### ISSUE TITLE
Gallery "add item" button is unlabeled

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#5823]
In the gallery block, the file upload button to add a gallery item doesn't have any text or aria-label so it's completely "empty" and can't be operated with assistive technologies. See https://github.com/WordPress/gutenberg/blob/af55057d2c6d19def327314f7efe8ad032ca4878/blocks/library/gallery/block.js#L248

![screen shot 2018-03-27 at 17 42 27](https://user-images.githubusercontent.com/1682452/37981300-9de30674-31ed-11e8-8d37-93260d65cc3b.jpg)

Any UI control needs to be labeled.

Since it happened several times that completely unlabeled elements were introduced in the UI, would it be feasible to use some specific tests? Specifically for buttons, is there any easy way to add tests to check if the buttons have some text or some non-empty aria-label to not allow empty buttons?

Worth reminding UI control names need to be visible or at least discoverable in some way. For example, speech recognition software users need to know the name of a control to be able to voice commands. For the icon buttons, Gutenberg uses the "tooltips": that's not 100% ideal, it's a compromise with design needs but at least there's a way to discover the control name.

In this specific case, given the size and the shape of this big button, I'm not sure a tooltip would look so nice. Adding some visible text seems to me the only solution.

**Comment by karmatosed** [comment]
Thanks for this. Noting here the design I suggested in PR is:

![](https://user-images.githubusercontent.com/253067/37984393-1213d78a-31ee-11e8-95d2-67bf7941b374.png)
